R/system.R

```diff
@@ -2019,8 +2035,24 @@ read_delim_file <- function(file, delim, quote = '"',
     if(stringi::stri_enc_mark(file) != "ASCII"){
       file_path <- file(file)
     }
-    readr::read_delim(file_path, delim, quote = quote, escape_backslash = escape_backslash, escape_double = escape_double, col_names = col_names, col_types = col_types,
-                      locale = locale, na = na, quoted_na = quoted_na, comment = comment, trim_ws = trim_ws, skip = skip, n_max = n_max, guess_max = guess_max, progress = progress)
+    tryCatch({ # try to close connection and ignore error
+      readr::read_delim(file_path, delim, quote = quote, escape_backslash = escape_backslash, escape_double = escape_double, col_names = col_names, col_types = col_types,
+                        locale = locale, na = na, quoted_na = quoted_na, comment = comment, trim_ws = trim_ws, skip = skip, n_max = n_max, guess_max = guess_max, progress = progress)
+    }, error = function(e) {
+      # For the case it's running on Linux (Collaboration Server), show more user friendly message.
+      # For Exploraotry Desktkop, it's already taken care of by Desktop so just show the error message as is.
+      if(Sys.info()["sysname"]=="Linux" && stringr::str_detect(stringr::str_to_lower(e$message), "invalid multibyte")) {
```

Comment:
Add comment on what was the full error message.

Reply:
Added comment.